### PR TITLE
Dxfeed  websocket integration tests

### PR DESCRIPTION
**Pull request by karen-stepanyan**
## Changes

- Added websocket integration tests for dxfeed and dxfeed-secondary EAs
- Added null safety checks for makeWSHandler [dxfeed]
- Modified `mockWebSocketFlow` function to accept options for enforcing mock order

## Steps to Test

` yarn test packages/sources/dxfeed/test/integration`
` yarn test packages/sources/dxfeed-secondary/test/integration
`
## Quality Assurance

- [x] Ran `yarn changeset` if adapter source code was changed
- [ ] If a new adapter was made, or an existing one was modified so that its environment variables have changed, update the relevant `<ADAPTER_PACKAGE>/schemas/env.json` and `<ADAPTER_PACKAGE>/README.md`
- [ ] If a new adapter was made, or an existing one was modified so that its environment variables have changed, update the relevant `infra-k8s` configuration file.
- [ ] The branch naming follows git flow (`feature/x`, `chore/x`, `release/x`, `hotfix/x`, `fix/x`) or is created from Clubhouse/Shortcut
- [ ] This is related to a maximum of one Clubhouse/Shortcut story or GitHub issue
- [x] Types are safe (avoid TypeScript/TSLint features like any and disable, instead use more specific types)

### CODE REVIEW

File: packages/sources/dxfeed-secondary/test/integration/adapter.test.ts [hunk 1]
```diff
@@ -4,16 +4,33 @@ import * as process from 'process'
...
+} from './fixtures'
+import {
+  mockWebSocketProvider,
+  mockWebSocketServer,
+  MockWsServer,
+  mockWebSocketFlow,
+} from '@chainlink/ea-test-helpers'
+import { WebSocketClassProvider } from '@chainlink/ea-bootstrap/dist/lib/middleware/ws/recorder'
+
+const sleep = (ms: number) => {
```
Comment (justinkaseman) [inline packages/sources/dxfeed-secondary/test/integration/adapter.test.ts:24]: You can use the new `sleep` util that Alejo abstracted to make this more DRY :man_dancing: 

```
import { util } from '@chainlink/ea-bootstrap'
await util.sleep(1000)
```

File: packages/core/test-helpers/src/websocket.ts [hunk 2]
```diff
@@ -40,41 +44,78 @@ export const MockWsServer = Server
...
+          // Message received matches expected request, remove exchange from the list
+          parsedFlow.shift()
         } else {
-          throw Error(
-            `The WS message received does not match the expected one. \nExpected: ${expected}\nReceived${received}`,
-          )
+          // If the sequence is not enforced, try to find the received msg within the list of expected exchanges
+          const i = parsedFlow.findIndex((exchange) => exchange.expected === received)
+
+          if (i === -1) {
```
Comment (austinborn) [inline packages/core/test-helpers/src/websocket.ts:103]: Personally I like using `.some()` better when the index itself doesn't matter:
```suggestion
          const messageExpected = parsedFlow.some((exchange) => exchange.expected === received)

          if (!messageExpected) {
```

File: packages/core/test-helpers/src/websocket.ts [hunk 3]
```diff
@@ -40,41 +44,78 @@ export const MockWsServer = Server
 export const mockWebSocketFlow = async (
   server: Server,
   flow: WsMessageExchange[],
+  options: {
+    enforceSequence: boolean
+    errorOnUnexpectedMessage: boolean
+  } = {
```
Comment (austinborn) [inline packages/core/test-helpers/src/websocket.ts:50]: Typescript can infer this type from the default
```suggestion
  options: {
```

File: packages/core/test-helpers/src/websocket.ts (hunk 2 above)
Comment (karen-stepanyan) [reply packages/core/test-helpers/src/websocket.ts:110]: it uses index few lines below to remove found element. If we get rid of index we would use something like filter which is less performant in this case

Comment (austinborn) [reply packages/core/test-helpers/src/websocket.ts:108]: Oh shoot, you're right. Ignore this comment...

**Review by austinborn — APPROVED** [review]

**Pull request merged**